From 474f9217339ce6ba46cf2001d829cc348e70f22d Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Mon, 24 Jan 2022 15:34:07 +0100
Subject: [PATCH 2/2] Recent versions of openldap have merged libldap_r into
 libldap

---
 src/utils/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/utils/CMakeLists.txt b/src/utils/CMakeLists.txt
index 2a7871b..1998f53 100644
--- a/src/utils/CMakeLists.txt
+++ b/src/utils/CMakeLists.txt
@@ -7,7 +7,8 @@ if (IS_IFCE)
     list (APPEND src_mds "${CMAKE_CURRENT_SOURCE_DIR}/mds/gfal_mds.c")
 else (IS_IFCE)
     list (APPEND gfal2_utils_definitions "-DMDS_BDII_EXTERNAL=0")
-    list (APPEND is_ifce_link "ldap_r" "lber")
+    find_library(LDAP_LIBRARY NAMES ldap_r ldap)
+    list (APPEND is_ifce_link ${LDAP_LIBRARY} "lber")
     list (APPEND src_mds
         "${CMAKE_CURRENT_SOURCE_DIR}/mds/gfal_mds.c"
         "${CMAKE_CURRENT_SOURCE_DIR}/mds/gfal_mds_internal.c"
-- 
2.29.2

